File: coloring.py
class Coloring:
    def __init__(self,n):
        # TODO

    def add_edge(self,a,b):
        # TODO

    def check(self):
        # TODO

if __name__ == "__main__":
    c = Coloring(4)
    c.add_edge(1,2)
    c.add_edge(2,3)
    c.add_edge(3,4)
    c.add_edge(1,4)
    print(c.check()) # True
    c.add_edge(2,4)
    print(c.check()) # False

File: Java_template/Coloring.java
public class Coloring {
    public Coloring(int n) {
        // TODO
    }

    public void addEdge(int a, int b) {
        // TODO
    }

    public boolean check() {
        // TODO
    }

    public static void main(String[] args) {
        Coloring c = new Coloring(4);
        c.addEdge(1,2);
        c.addEdge(2,3);
        c.addEdge(3,4);
        c.addEdge(1,4);
        System.out.println(c.check()); // true
        c.addEdge(2,4);
        System.out.println(c.check()); // false
    }
}
